lucille-core/src/main/java/com/kmwllc/lucille/util/CamelCaseConfigConverter.java

```diff
@@ -0,0 +1,125 @@
+package com.kmwllc.lucille.util;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.io.*;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.commons.text.CaseUtils;
+
+public class CamelCaseConfigConverter {
+
+  private static final Logger log = LoggerFactory.getLogger(CamelCaseConfigConverter.class);
+
+  private static String snakeToCamel(String snakeStr) {
+    return CaseUtils.toCamelCase(snakeStr, false, '_');
+  }
+
+  private static void writeToConfFile(String configString, String filePath) throws IOException {
+    try (PrintWriter writer = new PrintWriter(new FileWriter(filePath))) {
+      writer.printf(configString);
+      log.info("Written new config to {}", filePath);
+    } catch (Exception e) {
+      throw new IOException("Error writing new config to file path: " + filePath, e);
+    }
+  }
+
+  private static String applyCamelCase(File file, List<String> stageProperties) throws IOException {
+    StringBuilder sb = new StringBuilder();
+
+    try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        for (String stageProperty : stageProperties) {
+          if (line.contains(stageProperty)) {
+            log.debug("{}: {}", stageProperty, line);
+            line = processLine(line, stageProperty);
+          }
+        }
+        sb.append(line).append("\n");
+      }
+    } catch (Exception e) {
+      throw new IOException("Error processing file: " + file.getName(), e);
+    }
+
+    return sb.toString();
+  }
+
+  private static String processLine(String line, String stageProperty) throws Exception {
```

Comment:
what about String.replaceFirst()
if you want to test that there isn't a second instance of the stageProperty to replace, you could check that String.indexOf(stageProperty) is -1